File: 322. Coin Change.py
class Solution(object):
    def coinChange(self, coins, amount):
        """
        :type coins: List[int]
        :type amount: int
        :rtype: int
        """
        T = [-1 for i in xrange(amount+1)]
        T[0] = 0
        for a in xrange(amount+1):
            temp = amount + 1
            for c in coins:
                if c <= a and T[a-c] >= 0:
                    temp = min(temp, 1 + T[a-c])
            if temp < amount + 1:
                T[a] = temp
        return T[amount]

File: 322. Coin Change.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector <int> T (amount+1, -1);
        T[0] = 0;
        for (int a = 0; a <= amount; a++)
        {
            int temp = amount + 1;
            for (int c = 0; c < coins.size(); c++)
            {
                if (a >= coins[c] && T[a-coins[c]] >= 0)
                    temp = temp < T[a-coins[c]] + 1 ? temp : T[a-coins[c]] + 1;
            }
            if (temp < amount + 1)
                T[a] = temp;
        }
        return T[amount];
    }
};
